pandas/tests/frame/test_arithmetic.py

```diff
@@ -1147,6 +1147,33 @@ def test_arithmetic_midx_cols_different_dtypes_different_order(self):
         expected = DataFrame([[-1, 1], [-1, 1]], columns=midx)
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.xfail(reason="NaT op NaT results in datetime instead of timedelta")
```

Comment:
@jbrockmendel in this test, the comparison fails because the result dtype value of NaT is datetime64 and expected dtype value of NaT is Timedelta. I would expect NaT - NaT to result in Timedelta. Is this behavior expected?